### ISSUE TITLE
RFC: Your Proposal

### ISSUE THREAD

**Issue by glassp** [0no-co/gql.tada#394]
## Summary

Add a way to use environment variables for the schema url

## Proposed Solution

Allow to use environment variables within the plugins configuration
The plugin should also allow to load environment variables from an env file

Given the following tsconfig.json
```js

{
    "compilerOptions": {
        "plugins": [
            {
                "name": "gql.tada/ts-plugin",
                "environment": "./.env" // (optional) relative path to environment file
                "schemas": [
                    {
                        "name": "serviceA",
                        "schema": {
                            "url": "${SERVICE_A_BASE_URL}/delivery/graphql", // env usage with ${...}
                            "headers": {
                                "Accept": "application/graphql-response+json"
                            }
                        },
                        "tadaOutputLocation": "./src/GraphQL/ServiceA/graphql-env.d.ts",
                    },
                    {
                        "name": "serviceB",
                        "schema": {
                            "url": "${SERVICE_B_BASE_URL}/delivery/graphql", // env usage with ${...}
                            "headers": {
                                "Accept": "application/graphql-response+json"
                            }
                        },
                        "tadaOutputLocation": "./src/GraphQL/ServiceB/graphql-env.d.ts",
                    },
                ]
            }
        ]
    }
}

```

Without this feature switching between target environment e.g. local, integration, staging, production becomes a hazzle
Especially if hotfixing production directly we need to change the value in the .env and then again in the tsconfig which is often forgotten

**Comment by kitten** [comment]
There's some prior discussions on this on Discord and in a prior issue: https://github.com/0no-co/gql.tada/issues/324#issuecomment-2174277888

tl;dr: there's no point for the CLI and it won't make sense over established tools.

Furthermore, it's impossible in the `tsconfig.json` in general for editors, since your IDE does not have access necessarily to environment variables. This then necessitates injecting env-loading, which in turn is not a great idea.
